README.md
# Developer's Log

TYPO3 extension for logging calls to `GeneralUtility:devLog()` and browsing and searching those entries.
[Refer to the manual](https://docs.typo3.org/typo3cms/extensions/devlog/) for more details.


ChangeLog
2019-01-23 Francois Suter <typo3@cobweb.ch>

	* Catch uncompress errors when displaying extra data (thanks to E. Lang)
	* Ensure spinner icon is available
	* Mention enable_DLOG for TYPO3 v8 in the documentation
	* Release version 3.0.4

2018-02-22 Francois Suter <typo3@cobweb.ch>

	* Updated documentation configuration file
	* Released version 3.0.3

2018-02-18 Francois Suter <typo3@cobweb.ch>

	* Added check for database availability, resolves #6

2017-05-29 Francois Suter <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 8 LTS
	* Released version 3.0.2

2017-01-07 Francois Suter <typo3@cobweb.ch>

	* Added test plugin for generating log entries

2016-11-16 Francois Suter <typo3@cobweb.ch>

	* Added mapping for MarkJS, resolves #1
	* Fixed error due to possibly null pid, resolves #3
	* Released version 3.0.1

2016-11-15 Francois Suter <typo3@cobweb.ch>

	* Improved escaping of HTML content in log entries, resolves #2
	* Fixed error due to possibly null IP address (CLI context)

2016-11-13 Francois Suter <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 v8 (8.5.-dev), resolves #78626
	* Corrected awkward default value for IP filtering, resolves #78686
	* Released version 3.0.0

2016-11-09 Francois Suter <typo3@cobweb.ch>

	* Rewrote documentation and moved it to reST, resolves #78504
	* Fixed various bits and pieces, added Composer manifest, resolves #78624

2016-11-01 Francois Suter <typo3@cobweb.ch>

	* Updated and cleaned up the language files, resolves #78528

2016-10-31 Francois Suter <typo3@cobweb.ch>

	* Added an option to visually group rows by run, resolves #78058

2016-10-30 Francois Suter <typo3@cobweb.ch>

	* Restored and improved log clearing feature, resolves #76972
	* Added configuration option to restrict logging to certain keys, resolves #3053

2016-09-04 Francois Suter <typo3@cobweb.ch>

	* Restored column filters in new backend module, resolves #76971

2016-07-09 Francois Suter <typo3@cobweb.ch>

	* New backend module, resolves #76970

2016-07-06 Stefan Froemken <froemken@gmail.com>

	* Move extension configuration to domain model, resolves #73691

2014-12-24 Francois Suter <typo3@cobweb.ch>

	* Add sorting field, resolves #64037

2014-12-15 Francois Suter <typo3@cobweb.ch>

	* Allow for IP filtering, resolves #2859

2014-12-08 Francois Suter <typo3@cobweb.ch>

	* New table structure, new logging utility, database and file writer, resolves #63689
	* Labels for severity levels in extension configuration, resolves #31414

2014-05-21 Francois Suter <typo3@cobweb.ch>

	* Improved page id detection in FE context, resolves #58536
	* Released to TER as version 2.11.2

2014-03-24 Francois Suter <typo3@cobweb.ch>

	* Added check for empty BE user, resolves #57218
	* Released to TER as version 2.11.1

2014-03-22 Francois Suter <typo3@cobweb.ch>

	* Dropped NOT NULL constraint from data_var field definition, resolves #54702
	* Use dispatch mechanism for BE module
	* Add CSRF protection to all BE module forms and links, resolves #57194
	* Released to TER as version 2.11.0

2013-07-07 Francois Suter <typo3@cobweb.ch>

	* Removed old required files, resolves #49751 (thanks to Loek Hilgersom)

2013-05-28 Francois Suter <typo3@cobweb.ch>

	* Changed compatibility range to TYPO3 4.5-6.1
	* Fixed test of TYPO3_DB member variable
	* Fixed clearing of search and filter
	* Fixed filtering on namespaced keys
	* Fixed display of page icons, resolves #48605
	* Released to TER as version 2.10.0

2012-05-10 Francois Suter <typo3@cobweb.ch>

	* Fixed one forgotten compatibility issue, resolves #36957
	* Released to TER as version 2.9.4
	* Updated for compatibility with TYPO3 4.7, resolves #36957
	* Released to TER as version 2.9.3

2009-12-21 Francois Suter <typo3@cobweb.ch>

	* Added protection against possible XSS attacks
	* Released to TER as version 2.9.2

2009-09-19 Francois Suter <typo3@cobweb.ch>

	* Corrected TCA configuration so that devlog records are properly allowed anywhere in the page tree

2009-09-15 Francois Suter <typo3@cobweb.ch>

	* Removed some left-over debug statements (sorry!)
	* Released to TER as version 2.9.1

2009-09-09 Francois Suter <typo3@cobweb.ch>

	* Added optimization option when deleting excess records from the devlog table
	* Updated the manual
	* Released to TER as version 2.9.0

2009-08-17 Francois Suter <typo3@cobweb.ch>

	* Changed engine database for devlog table to InnoDB

2009-03-26 Francois Suter <typo3@cobweb.ch>

	* Updated the manual
	* Released to TER as version 2.8.0

2009-03-24 Francois Suter <typo3@cobweb.ch>

	* Added check for extra data type
	* Added configuration for filtering extra data of excessive size

2009-03-09 Francois Suter <typo3@cobweb.ch>

	* Added checks for proper initialization of DB object on devlog call
	* Added custom categories for extension configuration options (usable as of TYPO3 4.3)

2009-03-02 Francois Suter <typo3@cobweb.ch>

	* Made CSH for severity.image non translatable (thanks to Steffen Muller)
	* Completed update of the manual
	* Relased to TER as version 2.7.0

2009-03-01 Francois Suter <typo3@cobweb.ch>

	* Modified the CSH text, in particular severity levels descriptions
	* Updated the manual

2009-02-26 Francois Suter <typo3@cobweb.ch>

	* Completed implementation of CSH for both log table view and TCEform single view
	* Removed UID column in log table view
	* Moved date column to first place and placed links to log runs on that column
	* Changed default order in Web > List view, most recent record on top (thanks to Steffen Muller)
	* Changed log table styles slightly to better match t3skin

2009-02-24 Francois Suter <typo3@cobweb.ch>

	* Added CSH to header cells in log table view (thanks to Steffen Muller)

2009-02-16 Francois Suter <typo3@cobweb.ch>

	* Added better display of records in TCEforms (thanks to Fabien Udriot)
	* Added visible warning in BE module about filtered keys
	* Added t3lib_db::sql_free_results() calls wherever relevant
	* Added a button to clear search and all filters


ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext: "devlog"
#
# Auto generated 21-12-2009 22:35
#
# Manual updates:
# Only the data in the array - anything else is removed by next write.
# "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = [
        'title' => 'Developer\'s Log',
        'description' => 'The Developer\'s Log extension provides development logging/debugging functionality for usage of GeneralUtility::devlog() and a BE module to browse the logs.',
        'category' => 'misc',
        'priority' => '',
        'loadOrder' => '',
        'module' => 'mod1',
        'state' => 'stable',
        'uploadfolder' => 0,
        'createDirs' => '',
        'modify_tables' => '',
        'clearCacheOnLoad' => 1,
        'lockType' => '',
        'author' => 'Francois Suter',
        'author_email' => 'typo3@cobweb.ch',
        'author_company' => '',
        'version' => '3.0.4',
        'constraints' => [
                'depends' => [
                        'typo3' => '7.6.0-8.9.99',
                ],
                'conflicts' => [
                ],
                'suggests' => [
                ],
        ],
];